### PR TITLE
Add `author` and `keyword` search filters

### PR DESCRIPTION
**Pull request by Sherlouk**
This pull request introduces two new search filters. `author` and `keyword`. Both of these filters support equals to, and not equals to operators. Values are case insensitive, and keywords supports partial matching.

Based on discussions in #1403

http://localhost:8080/search?query=author%3ASherlouk

### CODE REVIEW

**Comment by Sherlouk** [comment]
@finestructure Am I missing something simpler with respect of doing a case insensitive comparison in this query? Or does my idea of adding a new field which is just a function on the existing values make sense?

**Comment by finestructure** [comment]
I'd use either `ilike`, like we do for the package page query:

```swift
            .filter(Repository.self, \.$owner, .custom("ilike"), owner)
            .filter(Repository.self, \.$name, .custom("ilike"), repository)
```

or if that isn't possible for some reason do a `lower(...)` on both sides of the comparison.

**Comment by Sherlouk** [comment]
Let me take a look into how the `where` query works and see if I can use the custom `ilike` comparison operator. Thanks for that pointer

**Comment by Sherlouk** [comment]
Thanks for that tip Sven, I've got it working as expected now!

http://localhost:8080/search?query=fluent+author%3AvaPor (fluent author:vaPor)
http://localhost:8080/search?query=fluent+author%3A%21vaPor (fluent author:!vaPor)

File: Sources/App/Core/Search.swift [hunk 1]
```diff
@@ -165,6 +165,7 @@ enum Search {
             .column(lastCommitDate)
             .column(lastActivityAt)
             .from(searchView)
+            .from(SQLFunction("CONCAT", args: keywords), as: keyword)
```
Comment (Sherlouk) [inline Sources/App/Core/Search.swift:168]: The output of this, for future reference is `{keyword,keyword2,keyword3}`.

**Review by daveverwer — APPROVED** [review]
This is great! Thank you James!

**Pull request merged**